packages/vite/src/build.ts

```diff
@@ -0,0 +1,96 @@
+import type { Plugin } from 'vite';
+import { join, dirname } from 'path';
+import { createHash } from 'crypto';
+import { fork } from 'child_process';
+import type { AddonMeta } from '@embroider/core';
+import { ResolverLoader } from '@embroider/core';
+import { existsSync, readdirSync, readFileSync, rmSync, writeFileSync, rmdirSync, copyFileSync } from 'fs';
+import { mkdirpSync } from 'fs-extra';
+
+const cwd = process.cwd();
+const embroiderDir = join(cwd, 'node_modules', '.embroider');
+const cacheKeyPath = join(embroiderDir, 'cache-key.json');
+
+export const lockFiles = ['pnpm-lock.yaml', 'package-lock.json', 'yarn.lock', 'package.json'];
+
+function getCacheKey(file: string) {
+  if (existsSync(cacheKeyPath)) {
+    return JSON.parse(readFileSync(cacheKeyPath).toString())[file];
+  }
+  return null;
+}
+
+function updateCacheKey(file: string, key: string | null) {
+  let json: Record<string, string | null> = {};
+  if (existsSync(cacheKeyPath)) {
+    json = JSON.parse(readFileSync(cacheKeyPath).toString());
+  }
+  json[file] = key;
+  writeFileSync(cacheKeyPath, JSON.stringify(json));
+}
+
+function computeCacheKeyForFile(file: string) {
+  if (existsSync(file)) {
+    const fileBuffer = readFileSync(file);
+    const hashSum = createHash('sha256');
+    hashSum.update(fileBuffer);
+
+    return hashSum.digest('hex');
+  }
+  return null;
+}
+
+export function emberBuild(): Promise<void> {
+  return new Promise((resolve, reject) => {
+    const child = fork('./node_modules/ember-cli/bin/ember', ['build']);
+    child.on('exit', code => (code === 0 ? resolve() : reject()));
+  });
```

Comment:
I don't think this is the best place to manage rebuilds of the addons. This kind of rebuild is extremely expensive compared with `ember s`.

We could still do the persistent caching you have here in order to avoid even a first-time build of the addons if none have changed since your last build. But that logic would go some place more like https://github.com/embroider-build/embroider/blob/3480aa65fa33b937748c9f94d8b334a1d52d8610/packages/compat/src/standalone-addon-build.ts#L13

Reply:
Yes,  well it would also be quite rare to do build. Only on package change and index.html change